src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EventPublisherContainer.java

```diff
@@ -16,6 +16,7 @@
  *     <li>use resource in {@link this#getPublisher(Classification, Supplier)})}</li>
  * </ul>
  */
+@SuppressWarnings("squid:S1612")
```

Comment:
lambdas vs method references

Reply:
my vote is for reference 🙃

Reply:
I went with lambda for readability reasons:
```
return caseDetails == null
    ? envelope -> exceptionRecordCreator.publish(envelope)
    : envelope -> attachDocsPublisher.publish(envelope, caseDetails.getCaseTypeId());
```
vs
```
return caseDetails == null
    ? exceptionRecordCreator::publish
    : envelope -> attachDocsPublisher.publish(envelope, caseDetails.getCaseTypeId());
```
in the 1st version we can see what's passed to the function and the similarities between the two functions.

Reply:
it reads better if it's consistent

Reply:
i remember the position, just stating my vote. all good and approved ;)